apps/src/code-studio/pd/workshop_survey/VariableFormGroup.jsx

```diff
@@ -0,0 +1,158 @@
+import React from 'react';
+
+import {ButtonList} from '../form_components/button_list';
+
+const SINGLE_SELECT = 'single_select';
+const MULTI_SELECT = 'multi_select';
+const FREE_RESPONSE = 'free_response';
+
+const questionPropType = React.PropTypes.shape({
+  label: React.PropTypes.string.isRequired,
+  name: React.PropTypes.string.isRequired,
+  required: React.PropTypes.bool,
+  type: React.PropTypes.oneOf([SINGLE_SELECT, MULTI_SELECT, FREE_RESPONSE]).isRequired,
+  values: React.PropTypes.arrayOf(React.PropTypes.string),
+});
+
+const ColumnVariableQuestion = React.createClass({
+  propTypes: {
+    selectedValues: React.PropTypes.arrayOf(React.PropTypes.string).isRequired,
+    question: questionPropType
+  },
+
+  buildColumn(selectedValue) {
+    const key = `${this.props.question.name}[${selectedValue}]`;
+
+    let type;
+    if (this.props.question.type === SINGLE_SELECT) {
+      type = 'radio';
+    } else if (this.props.question.type === MULTI_SELECT) {
+      type = 'check';
+    }
+
+    const formGroup = (
+      <ButtonList
+        answers={this.props.question.values}
+        groupName={key}
+        label={""}
+        type={type}
+      />
+    );
+
+    return (<td key={key}>{formGroup}</td>);
```

Comment:
[nope](https://github.com/airbnb/javascript/tree/master/react#alignment); short components are allowed to stay on a single line